src/object.rs

```diff
@@ -526,11 +526,20 @@ macro_rules! glib_object_wrapper {
             #[cfg_attr(feature = "cargo-clippy", allow(cast_ptr_alignment))]
             unsafe fn from_glib_borrow(ptr: *mut $ffi_name) -> Self {
                 debug_assert!($crate::types::instance_of::<Self>(ptr as *const _));
-                $name($crate::translate::from_glib_borrow(ptr as *mut _),
+                $name($crate::translate::from_glib_borrow(ptr as *mut $ffi_name),
```

Comment:
`ptr: *mut $ffi_name`  converted to `*mut $ffi_name` ? seems previous it was something else, IMHO ObjectRef need `*mut $crate::gobject_ffi::GObject` but not sure .

Reply:
Yes, the whole line needs to be `$crate::translate::from_glib_borrow(ptr as *const $ffi_name)`. No need to put `$name(...)` around it all.

Reply:
I replaced with `as _`. Seems better.

Reply:
Why? You need a `*mut $ffi_name` here and not some random pointer. It's safer to be explicit here

Reply:
Now it is with `$name(...)` and the cast to `*mut $ffi_type`, and that's the only combination that does not work.

Reply:
Locally I get this error:
```
error[E0277]: the trait bound `object::ObjectRef: translate::FromGlibPtrBorrow<*mut gobject_ffi::GBinding>` is not satisfied
   --> D:/eap/rust/0/glib\src\object.rs:529:23
    |
529 |                   $name($crate::translate::from_glib_borrow(ptr as *mut $ffi_name),
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `translate::FromGlibPtrBorrow<*mut gobject_ffi::GBinding>` is not implemented for `object::ObjectRef`
    | 
   ::: D:/eap/rust/0/glib\src\gobject\auto\binding.rs:12:1
    |
12  | / glib_wrapper! {
13  | |     pub struct Binding(Object<ffi::GBinding, BindingClass>);
14  | |
15  | |     match fn {
16  | |         get_type => || ffi::g_binding_get_type(),
17  | |     }
18  | | }
    | |_- in this macro invocation
    |
    = help: the following implementations were found:
              <object::ObjectRef as translate::FromGlibPtrBorrow<*const gobject_ffi::GObject>>
              <object::ObjectRef as translate::FromGlibPtrBorrow<*mut gobject_ffi::GObject>>
note: required by `translate::from_glib_borrow`
   --> D:/eap/rust/0/glib\src\translate.rs:1052:1
    |
1052| pub unsafe fn from_glib_borrow<P: Ptr, T: FromGlibPtrBorrow<P>>(ptr: P) -> T {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Yes, that's exactly what I'm saying :) Remove the `$name(...)` part and it will work

Reply:
Removing name IMHO don't help as this is definition from_glib_borrow(*mut $ffi_name) that used by `::from_glib_borrow`

Reply:
`*mut $crate::gobject_ffi::GObject` IMHO solve problem

Reply:
We might be talking about different functions. I mean the implementation for `*const $ffi_name` that should just call into the one for `*mut $ffi_name`.

The implementation for `*mut $ffi_name` needs the `$name(...)` wrapper and has to be cast to `*mut GObject`.